packages/core/rollup.utils.js

```diff
@@ -0,0 +1,167 @@
+        {
+          writeBundle(outputOptions, bundle) {
+            const css = Object.entries(bundle)[1][1].source;
+            fs.writeFileSync(outputOptions.file, css);
+            fs.writeFileSync(outputOptions.file.replace('.css', '.min.css'), csso.minify(css).css);
+          },
+          buildStart() {
+            this.addWatchFile(input);
+          },
+        },
+      ],
+    };
+  });
+};
+
+async function treeshakeCSS(content, css) {
+  const value = await new PurgeCSSDefault.PurgeCSS().purge({
+    content,
+    defaultExtractor: content => content.match(/[\w-\/:@]+(?<!:)/g) || [],
+    safelist: [/:host$/, /:slotted$/],
+    css: [{ raw: css }],
+    variables: true,
+  });
+  return value[0].css;
+}
+
+export const addComponentEntryPoints = (packageFile, config) => {
+  // https://lit.dev/docs/tools/publishing/
+  // https://justinfagnani.com/2019/11/01/how-to-publish-web-components-to-npm
+  // https://nodejs.org/api/packages.html#packages_subpath_exports
+
+  const explicitExports = config.entryPoints.explicitExports.map(m => `"${m.input}": "${m.output}"`);
+
+  const modules = config.entryPoints.modules.map(m => {
+    const path = `.${resolve(m).replace(resolve(config.baseDir), '')}/index.js`;
+    return `"${dirname(path)}": "${path}"`;
+  });
+
+  const components = config.entryPoints.components.flatMap(m => {
+    const path = `.${resolve(m).replace(resolve(config.baseDir), '')}`;
+    return [
+      `"${path}": "${path}/index.js"`,
+      `"${path}/register": "${path}/register.js"`,
+      `"${path}/register.js": "${path}/register.js"`,
+    ];
+  });
+
+  const styles = config.entryPoints.styles.flatMap(m => {
+    const output = typeof m === 'string' ? m : m.output;
+    const path = `.${resolve(output)
+      .replace(resolve(config.baseDir), '')
+      .replace(resolve(config.outDir), '')
+      .replace(extname(output), '.css')}`;
+    const minPath = path.replace(extname(path), '.min.css');
+    return [`"${path}": "${path}"`, `"${minPath}": "${minPath}"`];
+  });
+
+  const exports = JSON.parse(`{
+    "./package.json": "./package.json",
+    "./custom-elements.json": "./custom-elements.json",
+    ${[modules, explicitExports, components, styles].join(',')}
+  }`);
+
+  const sideEffects = config.entryPoints.components.map(
+    name => `.${resolve(name).replace(resolve(config.baseDir), '')}/register.js`
+  );
+
+  return JSON.stringify({ ...JSON.parse(packageFile), sideEffects, exports }, null, 2);
+};
+
+const fileCache = {};
+
+export const esmCache = outDir => {
+  return {
+    name: 'esm-cache',
+    generateBundle(_options, bundles) {
+      for (const [key, bundle] of Object.entries(bundles)) {
+        const path = `${outDir}/${bundle.fileName}`;
+
+        if (extname(path) === '.js' && fileCache[path] !== bundle.code) {
+          fileCache[path] = bundle.code;
+        } else {
+          delete bundles[key];
+        }
+      }
+    },
+  };
+};
```

Comment:
Rollup plugins/utils for the Core build. A follow up PR will make these better encapsulated for the clarity addon support.